Rsz hier

andyfox-rushc opened the pull request:
Preliminary work on resizer on hierarchical db. This pull request covers repairHold only.
Unit test repair_hold1_hier added.

maliberty commented:
is this ready for review?

andyfox-rushc commented:
Hi, Matt,
This is ready for review.  Please note I am thinking through apis for using the hierarchy calls  and will update the dbNetwork to include a cleaned up version after doing the RepairSetup once I know more about the apis needed. For example the buffer creation stuff in rsz uses an Instance* as the parent scope (see create buffer code) which feels wrong ! I think it best to get through the RepairSetup part and then rework the code with some cleaner apis so please note there are a lot of low level dbSta/Stadb calls which need to be pushed into api calls in dbNetwork.cc, once we know what those calls should be.

QuantamHD commented on src/rsz/src/Resizer.cc:
```diff
@@ -3319,6 +3319,25 @@ void Resizer::journalRestoreTest()
+  ip = nullptr;
+  op = nullptr;
+
+  InstancePinIterator* pin_iter = network_->pinIterator(buffer);
```
You should use std::unique_ptr to mange iterators.

```c++
auto pin_iter = std::unique_ptr<InstancePinIterator>(network_->pinIterator(buffer))
```

maliberty commented on src/odb/include/odb/db.h:
```diff
@@ -3561,9 +3561,14 @@ class dbITerm : public dbObject
   void disconnect();
 
+  ///
+  /// Selectively disconnect db_net/mod_net
+  ///
+  void disconnect(bool db_net, bool mod_net);
```
I'm not fond of this API as it leaves the db in an inconsistent state.  I would rather both were connected/disconnected together.

andyfox-rushc replied:
removed

andyfox-rushc commented:
Matt, I addressed all the comments posted, let me know if there is more needed to have this minor pull request accepted. The setup fixing is clearly going to be an order of magnitude more challenging for review.
